pkg/frontend/nvme.go

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Intel Corporation
+
+// Package frontend implements the FrontEnd APIs (host facing) of the storage Server
+package frontend
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	pb "github.com/opiproject/opi-api/storage/v1alpha1/gen/go"
+	"github.com/opiproject/opi-spdk-bridge/pkg/models"
+	"github.com/ulule/deepcopier"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const (
+	mevTransportType = "npi"
+)
+
+type mevNvmfSubsystemAddListenerParams struct {
```

Comment:
this is standard SPDK struct, let's move it to https://github.com/opiproject/opi-spdk-bridge/blob/main/pkg/models/spdk.go

Reply:
I mean how this is different from 
```
// NvmfSubsystemAddListenerParams holds the parameters required to Delete a NVMf subsystem
type NvmfSubsystemAddListenerParams struct {
	Nqn           string `json:"nqn"`
	ListenAddress struct {
		Trtype  string `json:"trtype"`
		Traddr  string `json:"traddr"`
		Trsvcid string `json:"trsvcid"`
		Adrfam  string `json:"adrfam"`
	} `json:"listen_address"`
}
``` 
?

Reply:
maybe you just wanted to avoid sending extra fields? please cosider then using `,omitempty`
for example:
```
type NvmfSubsystemAddListenerParams struct {
	Nqn           string `json:"nqn"`
	ListenAddress struct {
		Trtype  string `json:"trtype"`
		Traddr  string `json:"traddr"`
		Trsvcid string `json:"trsvcid,omitempty"`
		Adrfam  string `json:"adrfam,omitempty"`
	} `json:"listen_address"`
}
```

Reply:
Yes, you are correct. I saw that structure, but it didn't fit our needs, since Trsvcid and Adrfam should be omitted. With this patch, I wanted to initiate a discussion if we can change the structure to the way you proposed. As I understand, you do not see any issues, right?

Reply:
Yes, I'm perfectly OK with adding `,omitempty` on the original struct and then you can just use it here

Reply:
updated